sage.py

```diff
@@ -104,105 +89,76 @@ def load_IANA_mapping():
                     "name": row[0] if row[0] else "unknown",
                     "description": row[3] if row[3] else "---",
                 }
-        else:
-            # Do nothing 
-            pass
     return ports
-    
-## 3
-def readfile(fname):
-    unparsed_data = None
+
+
+def _readfile(fname):
     with open(fname, 'r') as f:
         unparsed_data = json.load(f)
         
     unparsed_data = unparsed_data[::-1]
-    #print('# events: ', len(unparsed_data))
-    #print(unparsed_data[0])
     return unparsed_data
-    
 
-#cats = dict()
-#ips = dict()
-#hosts = dict()
-#h_trig = []
 
-def parse(unparsed_data, alert_labels=[], slim=False):
-    
+# Step 1.1: Parse the input alerts
+def _parse(unparsed_data, alert_labels=[], slim=False):
     FILTER = False
     badIP = '169.254.169.254'
     __cats = set()
     __ips = set()
     __hosts = set()
-    __sev = set()
-    data = []
-    connections = dict()
+    parsed_data = []
 
     prev = -1
-    for id, d in enumerate(unparsed_data):
-        #print(d)
-        raw = ''
-        
-        try:
-             raw = json.loads(d['result']['_raw'])
-        except:
-            try:
-                raw = json.loads(d['_raw'])
-            except:
-                raw = d
-                
+    for d in unparsed_data:
+        if 'result' in d and '_raw' in d['result']:
+            raw = json.loads(d['result']['_raw'])
+        elif '_raw' in d:
+            raw = json.loads(d['_raw'])
+        else:
+            raw = d
+
         if raw['event_type'] != 'alert':
             continue
-        #app_proto = raw['app_proto']
-        host = ''
-        
-        try:
+
+        if 'host' in raw:
             host = raw['host']
-        except:
-            try:
-                host = d['host'][3:]
-            except:
-                host = 'dummy'
-                
+        elif 'host' in d:
+            host = d['host'][3:]
+        else:
+            host = 'dummy'
 
-        #print(host)
-        ts = raw['timestamp']
-        dt = datetime.datetime.strptime(ts, '%Y-%m-%dT%H:%M:%S.%f%z')# 2018-11-03T23:16:09.148520+0000
-        DIFF = 0.0 if prev== -1 else round((dt - prev).total_seconds(),2)
```

Comment:
PEP was complaining about capitalised local variables, so I have changed the name